### PR TITLE
add codes for tuple gauss sieve

### PR DESCRIPTION
**Pull request by shi-bai**

### CODE REVIEW

File: src/sieve/sampler_basic.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  for (int i = 0; i < nr; i++) {
+    tmp.set_z(g(i,i));
+    ((*s_prime)[i]).div(maxbistar2, tmp);
+    ((*s_prime)[i]).sqrt((*s_prime)[i], GMP_RNDN);
+  }
+
+  /* verbose */
+  set_verbose(ver);
+  print_param();
+  gmp_randinit_default(state);
```
Comment (malb) [inline src/sieve/sampler_basic.cpp]: Why does the sampler maintain its own state? Wouldn't it be better to use the common state for reproducibility?

File: src/sieve/sampler_basic.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  private:
+
+  /**
+   * GSO object of B
+   */
+  int nr, nc;
+  double logn2;
+  gmp_randstate_t state;
+  ZZ_mat<ZT> b, u, uInv;
+  MatGSO<Z_NR<ZT>, F> *pGSO;
```
Comment (malb) [inline src/sieve/sampler_basic.h]: My C++ isn't strong enough but couldn't this just be a MatGSO object (not a pointer)?

File: src/sieve/sampler_main.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
+#include "sampler_basic.h"
```
Comment (malb) [inline src/sieve/sampler_main.cpp]: Shouldn't this live in tests?

File: src/sieve/sieve_Gauss.cpp [hunk 4]
```diff
@@ -0,0 +1,323 @@
+/*
```
Comment (malb) [inline src/sieve/sieve_Gauss.cpp]: maybe sieve_gauss.cpp?

**Comment by malb** [comment]
Also, it seems the sieving code isn't actually built.

File: src/sieve/sieve_Gauss.cpp (hunk 4 above)
Comment (shi-bai) [inline src/sieve/sieve_Gauss.cpp]: done, fixed above. It seems compilting now with
make latsieve
there was some change in the codebase I guess which made it broken :)

**Comment by shi-bai** [comment]
delete this pull request as my fork has been removed previously..

**Pull request closed without merging**